internal/compiler/passes/resolving.rs

```diff
@@ -1049,11 +1066,19 @@ fn continue_lookup_within_element(
         }
         Expression::CallbackReference(NamedReference::new(elem, &lookup_result.resolved_name))
     } else if matches!(lookup_result.property_type, Type::Function { .. }) {
+        if let Some(x) = it.next() {
+            ctx.diag.push_error("Cannot access fields of a function".into(), &x)
+        }
         let member = elem.borrow().base_type.lookup_member_function(&lookup_result.resolved_name);
-        Expression::MemberFunction {
-            base: Box::new(Expression::ElementReference(Rc::downgrade(elem))),
-            base_node: Some(NodeOrToken::Node(node.into())),
-            member: Box::new(member),
+        if !matches!(member, Expression::Invalid) {
+            // builtin member function
+            Expression::MemberFunction {
+                base: Box::new(Expression::ElementReference(Rc::downgrade(elem))),
+                base_node: Some(NodeOrToken::Node(node.into())),
+                member: Box::new(member),
+            }
+        } else {
+            Expression::CallbackReference(NamedReference::new(elem, &lookup_result.resolved_name))
```

Comment:
I looked a bit at the uses of `Expression::CallbackReference` and I think adding `Expression::FunctionReference` is overall cleaner. (This is a matter of taste, of course)

Either way it requires careful checking of all match sites. I noticed that `resolve_two_way_binding` needs to be adjusted as well, to produce the correct error message when trying this:

```
    property <string> foo <=> bar;
    function bar() {}
```

(should produce "Cannot bind to a function" instead of "Cannot bind to a callback")